homeassistant/components/aidot/light.py

```diff
@@ -0,0 +1,268 @@
+        """Return True if the light is on."""
+        return self.lanCtrl.is_on
+
+    @property
+    def brightness(self) -> int:
+        """Return the brightness of this light between 0..255."""
+        return self.lanCtrl.brightness
+
+    @property
+    def min_color_temp_kelvin(self) -> int:
+        """Return the warmest color_temp_kelvin that this light supports."""
+        return self._cct_min
+
+    @property
+    def max_color_temp_kelvin(self) -> int:
+        """Return the coldest color_temp_kelvin that this light supports."""
+        return self._cct_max
+
+    @property
+    def color_temp_kelvin(self) -> int | None:
+        """Return the CT color value in Kelvin."""
+        return self.lanCtrl.cct
+
+    @property
+    def rgbw_color(self) -> tuple[int, int, int, int] | None:
+        """Return the rgbw color value [int, int, int, int]."""
+        if self.lanCtrl.rgdb:
+            rgbw = ctypes.c_uint32(self.lanCtrl.rgdb).value
+            r = (rgbw >> 24) & 0xFF
+            g = (rgbw >> 16) & 0xFF
+            b = (rgbw >> 8) & 0xFF
+            w = rgbw & 0xFF
+            return (r, g, b, w)
+        return self._attr_rgbw_color
+
+    @property
+    def color_mode(self) -> ColorMode | str | None:
+        """Return the color mode of the light."""
+        if self.lanCtrl.colorMode == Attribute.RGBW:
+            colorMode = ColorMode.RGBW
+        elif self.lanCtrl.colorMode == Attribute.CCT:
+            colorMode = ColorMode.COLOR_TEMP
+        else:
+            colorMode = ColorMode.BRIGHTNESS
+        return colorMode
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the light on."""
+        brightness = kwargs.get(ATTR_BRIGHTNESS, 255)
+
+        if self.lanCtrl.connectAndLogin is False:
+            _LOGGER.error(
+                "The device is not logged in or may not be on the local area network"
+            )
+            raise HomeAssistantError(
+                "The device is not logged in or may not be on the local area network"
+            )
+
+        action = {}
+        if ATTR_BRIGHTNESS in kwargs and brightness is not None:
+            action.update(self.lanCtrl.getDimingAction(brightness))
+        if ATTR_COLOR_TEMP_KELVIN in kwargs:
+            cct = kwargs.get(ATTR_COLOR_TEMP_KELVIN)
+            action.update(self.lanCtrl.getCCTAction(cct))
+        if ATTR_RGBW_COLOR in kwargs:
+            rgbw = kwargs.get(ATTR_RGBW_COLOR)
+            if rgbw is not None:
+                rgbw = (rgbw[0] << 24) | (rgbw[1] << 16) | (rgbw[2] << 8) | rgbw[3]
+                ctype_result = ctypes.c_int32(rgbw).value
+                action.update(self.lanCtrl.getRGBWAction(ctype_result))
+        if not kwargs:
+            action.update(self.lanCtrl.getOnOffAction(1))
+
+        await self.lanCtrl.sendDevAttr(action)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the light off."""
+        if self.lanCtrl.connectAndLogin is False:
```

Comment:
We have the `available` entity property that should be set to `False` if the device is offline. Then we don't need to check in the service action handlers. The CoordinatorEntity has a default `available` property that can be overridden as needed.